core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -307,6 +310,8 @@ private QueryFetchSearchResult executeFetchPhase(SearchContext context, SearchOp
 
     public ScrollQuerySearchResult executeQueryPhase(InternalScrollSearchRequest request, SearchTask task) {
         final SearchContext context = findContext(request.id());
+        if(context == null)
+            return null;
```

Comment:
I don't like introducing such a condition, could it instead return a proper `ScrollQuerySearchResult` instance, like it would have got before?